tests/tools/visualization/test_graph.py

```diff
@@ -353,6 +353,7 @@ def test_graph_node_identifiers(self, node_id_type, monkeypatch, file_regression
         # The order of certain output lines can be randomly ordered so we split the file in lines, sort, and then join
         # them into a single string again. The node identifiers generated by the engine are of the form ``N{pk}`` and
         # they also clearly vary, so they are replaced with the ``NODE`` placeholder.
-        string = '\n'.join(sorted(graph.graphviz.source.strip().split('\n')))
```

Comment:
If the sort happened first, then the order may depend on the pk number and the test fail with different string where lines order changed. I sort the lines after replace the pk number.